homeassistant/components/luxtronik/__init__.py

```diff
@@ -0,0 +1,157 @@
+"""Support for Luxtronik heatpump controllers."""
+# region Imports
+from luxtronik import LOGGER as LuxLogger
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.typing import ConfigType
+
+from .const import (
+    CONF_LANGUAGE_SENSOR_NAMES,
+    CONF_LOCK_TIMEOUT,
+    CONF_SAFE,
+    DOMAIN,
+    LANG_DEFAULT,
+    LOGGER,
+    LUX_SENSOR_DETECT_COOLING,
+    PLATFORMS,
+)
+from .helpers.helper import get_sensor_text
+from .helpers.lux_helper import get_manufacturer_by_model
+from .luxtronik_device import LuxtronikDevice
+
+# endregion Imports
+
+# region Constants
+LuxLogger.setLevel(level="WARNING")
+# endregion Constants
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Set up from config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    LOGGER.info(
+        "%s.async_setup_entry options: '%s' data:'%s'",
+        DOMAIN,
+        config_entry.options,
+        config_entry.data,
+    )
+    config_entry.async_on_unload(config_entry.add_update_listener(async_reload_entry))
+
+    setup_internal(hass, config_entry.data, config_entry.options)
+
+    luxtronik = hass.data[DOMAIN]
+
+    hass.config_entries.async_setup_platforms(config_entry, PLATFORMS)
+
+    def logout_luxtronik(event: Event) -> None:
+        """Close connections to this heatpump."""
+        luxtronik.disconnect()
+
+    config_entry.async_on_unload(
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, logout_luxtronik)
+    )
+    return True
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the luxtronik component."""
+    if DOMAIN not in config:
+        # Setup via UI. No need to continue yaml-based setup
```

Comment:
This is a new integration, we should not have YAML based configuration.
The `async_setup` method can be fully removed.